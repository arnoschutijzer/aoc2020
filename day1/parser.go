package dayone

import (
	"io/ioutil"
	"strconv"
	"strings"
)

func ReadFile(path string) []int {
	data, _ := ioutil.ReadFile(path)
	textinput := string(data)
	list := strings.Split(textinput, "\n")

	var expenses []int
	for _, expense := range list {
		parsedExpense, _ := strconv.Atoi(expense)
		expenses = append(expenses, parsedExpense)
	}

	return expenses
}
