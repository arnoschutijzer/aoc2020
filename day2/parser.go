package daytwo

import (
	"io/ioutil"
	"regexp"
	"strconv"
	"strings"
)

func ReadFile(path string) []Password {
	data, _ := ioutil.ReadFile(path)
	textinput := string(data)
	list := strings.Split(textinput, "\n")

	var passwords []Password

	for _, passwordEntry := range list {
		var password Password
		re := regexp.MustCompile(`([0-9])*-(0-9)*\w+`)
		characterCount := re.FindString(passwordEntry)

		minOccurence, _ := strconv.Atoi((string(characterCount[0])))
		maxOccurence, _ := strconv.Atoi((string(characterCount[2])))
		password.minOccurence = minOccurence
		password.maxOccurence = maxOccurence

		re = regexp.MustCompile(`([A-z]):+`)
		password.letter = strings.ReplaceAll(re.FindString(passwordEntry), ":", "")

		seperatorIndex := strings.Index(passwordEntry, ": ")
		password.password = string(passwordEntry[seperatorIndex+2:])

		passwords = append(passwords, password)
	}

	return passwords
}
