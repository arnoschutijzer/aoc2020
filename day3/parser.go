package daythree

import (
	"io/ioutil"
	"strings"
)

func ParseTopology(path string) [][]string {
	bytes, _ := ioutil.ReadFile(path)
	input := string(bytes)

	rows := strings.Split(input, "\n")

	var topology [][]string

	for _, value := range rows {
		topology = append(topology, strings.Split(value, ""))
	}

	return topology
}
