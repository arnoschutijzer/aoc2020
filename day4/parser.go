package dayfour

import (
	"io/ioutil"
	"regexp"
	"strings"
)

func ParsePassports(path string) []Passport {
	bytes, _ := ioutil.ReadFile(path)
	contents := string(bytes)

	passportStrings := strings.Split(contents, "\n\n")

	var passports []Passport
	for _, passportString := range passportStrings {
		passports = append(passports, Passport{
			fields: passportString,
			pid:    GetIDFromPassword(passportString),
		})
	}

	return passports
}

func GetIDFromPassword(stringifiedPassport string) string {
	re := regexp.MustCompile(`(?m)pid:(?P<value>[0-9]{9})($| )`)
	matches := re.FindStringSubmatch(stringifiedPassport)

	indexOfSub := re.SubexpIndex("value")
	if indexOfSub >= len(matches) {
		return ""
	}

	return matches[re.SubexpIndex("value")]
}
