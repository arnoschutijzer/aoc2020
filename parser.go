package parser

import (
	"io/ioutil"
	"strings"
)

func ReadFile(path string) []string {
	data, _ := ioutil.ReadFile(path)
	textinput := string(data)
	list := strings.Split(textinput, "\n")

	return list
}
