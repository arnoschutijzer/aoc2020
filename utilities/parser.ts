import fs from 'fs';
import path from 'path';

export function readFile(filePath: string): string[] {
  const contents = fs.readFileSync(path.resolve(filePath)).toString();
  return contents.split("\n");
}