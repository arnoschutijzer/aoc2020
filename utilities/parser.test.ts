import { readFile } from './parser'

describe('parser', () => {
  it('parses some_file.txt', () => {
    expect(readFile(`${__dirname}/fixtures/some_file.txt`)).toMatchSnapshot();
  });
});