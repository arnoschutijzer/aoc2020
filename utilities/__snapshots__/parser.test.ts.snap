// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser parses some_file.txt 1`] = `
Array [
  "this",
  "file",
  "contains",
  "some",
  "text",
]
`;
